src/store/auth/sagas.ts

```diff
@@ -1,29 +1,151 @@
+import { getProjectIdResult } from '../config/selectors';
+import {
+  authFetchUsersError,
+  authFetchUsersRequest,
+  authFetchUsersSuccess,
+  createUserRequest,
+  createUserSuccess,
+  deleteUserRequest,
+  deleteUserSuccess,
+  getAllowDuplicateEmailsRequest,
+  nukeUsersRequest,
+  nukeUsersSuccess,
+  setAllowDuplicateEmailsRequest,
+  setAllowDuplicateEmailsSuccess,
+  setUserDisabledRequest,
+  setUserDisabledSuccess,
+  updateUserRequest,
+  updateUserSuccess,
+} from './actions';
+import { getAuthConfigResult } from './selectors';
+
+const AUTH_API_CONTEXT_KEY = 'authApi';
+
+export function* fetchAuthUsers() {
+  const authApi = yield call(configureAuthSaga);
+  try {
+    const users = yield call([authApi, 'fetchUsers']);
+    yield put(authFetchUsersSuccess(users));
+  } catch (e) {
+    yield put(authFetchUsersError({ message: 'Error when fetching users' }));
+  }
+}
+
+export function* deleteUser({ payload }: ReturnType<typeof deleteUserRequest>) {
+  const authApi = yield call(configureAuthSaga);
+  yield call([authApi, 'deleteUser'], { localId: payload.localId });
+  yield put(deleteUserSuccess({ localId: payload.localId }));
+}
+
+export function* createUser({ payload }: ReturnType<typeof createUserRequest>) {
+  const authApi = yield call(configureAuthSaga);
+  const newUser: AuthUser = yield call([authApi, 'createUser'], payload.user);
+  const user = {
+    ...newUser,
+    ...payload.user,
+  };
+
+  if (payload.user.customAttributes) {
+    yield call([authApi, 'updateUser'], user);
+  }
+  yield put(
+    createUserSuccess({
+      user,
+    })
+  );
+}
+
+export function* updateUser({ payload }: ReturnType<typeof updateUserRequest>) {
+  const authApi = yield call(configureAuthSaga);
+
+  const user: AuthUser = yield call([authApi, 'updateUser'], {
+    ...payload.user,
+    localId: payload.localId,
+  });
+  yield put(updateUserSuccess({ user: { ...user, ...payload.user } }));
+}
+
+export function* setUserDisabled({
+  payload,
+}: ReturnType<typeof setUserDisabledRequest>) {
+  const authApi = yield call(configureAuthSaga);
+  yield call([authApi, 'updateUser'], {
+    disableUser: payload.disabled,
+    localId: payload.localId,
+  });
+  yield put(setUserDisabledSuccess(payload));
+}
+
+export function* setAllowDuplicateEmails({
+  payload,
+}: ReturnType<typeof setAllowDuplicateEmailsRequest>) {
+  const authApi = yield call(configureAuthSaga);
+  yield call([authApi, 'updateConfig'], payload);
+  yield put(setAllowDuplicateEmailsSuccess(payload));
+}
+
+export function* getAllowDuplicateEmails() {
+  const authApi = yield call(configureAuthSaga);
+  const config = yield call([authApi, 'getConfig']);
+  yield put(setAllowDuplicateEmailsSuccess(config));
+}
+
+export function* nukeUsers() {
+  const authApi = yield call(configureAuthSaga);
+  yield call([authApi, 'nukeUsers']);
+  yield put(nukeUsersSuccess());
+}
+
+export function* configureAuthSaga() {
+  let authApi = yield getContext(AUTH_API_CONTEXT_KEY);
```

Comment:
Please don't cache this -- this will fail if the emulator suite gets shutdown and restarted with Auth Emulator running on a different port / different project ID. This doesn't really give us much performance improvements anyway since the API object is really cheap to construct.